helixlauncher-core/src/auth/account.rs

```diff
@@ -15,34 +15,42 @@ pub enum AccountManagerError {
     IoError(#[from] std::io::Error),
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct Account {
     pub uuid: String,
     pub username: String,
     pub refresh_token: String,
     pub token: String,
-    pub selected: bool,
 }
 
-pub fn get_accounts(account_json: &Path) -> Result<Vec<Account>, AccountManagerError> {
-    // try to open file, return empty if file doesnt exist
-    Ok(serde_json::from_reader(BufReader::new(
-        match File::open(account_json) {
-            Err(err) if err.kind() == io::ErrorKind::NotFound => return Ok(vec![]),
-            result => result,
-        }?,
-    ))?)
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct AccountConfig {
+    pub accounts: Vec<Account>,
+    /// The UUID of the selected account
+    pub selected: String,
+    #[serde(skip)]
+    path: PathBuf,
 }
 
-pub fn add_account(mut account: Account, account_json: &Path) -> Result<(), AccountManagerError> {
-    let mut accounts: Vec<Account> = get_accounts(account_json)?;
-    if accounts.is_empty() {
-        account.selected = true
```

Comment:
This feature appears to be missing in the new implementation: defaulting to the single account in the list if there is only one (though that should probably be done in the place it's used in, if the selected default is `None`)